tephra-core/src/main/java/org/apache/tephra/TxConstants.java

```diff
@@ -158,6 +158,14 @@
      * The default value for the transaction timeout limit, in seconds: unlimited.
      */
     public static final int DEFAULT_TX_MAX_TIMEOUT = Integer.MAX_VALUE;
+    /**
+     * The maximum time in seconds that a transaction can be used for data writes.
```

Comment:
Does this apply to all transaction, including long-running ones? I think it helps to say that after this time, all writes to the table will fail, and that is ensured by Tephra's coprocessors.

You should also document that this value influences invalid tx pruning.